### PR TITLE
[WebProfilerBundle] Move xml service configuration to php

### PR DESCRIPTION
**Pull request by jschaedl**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no 
| Deprecations? | no 
| Tickets       | #37186  
| License       | MIT
| Doc PR        | -

Move WebProfilerBundles xml service configuration to php.

### CODE REVIEW

**Review by stof — APPROVED** [review]

**Comment by fabpot** [comment]
I think you need to add a conflict with dependency-injection < 5.2 so that tests pass on low.

File: src/Symfony/Bundle/WebProfilerBundle/Resources/config/profiler.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                    ->args([null, param('kernel.charset'), HtmlDumper::DUMP_LIGHT_ARRAY])
+                    ->call('setDisplayOptions', [['maxStringLength' => 4096, 'fileLinkFormat' => service('debug.file_link_formatter')]]),
+            ])
+
+        ->set('debug.file_link_formatter', FileLinkFormatter::class)
+            ->args([
+                param('debug.file_link_format'),
+                service('request_stack')->ignoreOnInvalid(),
+                param('kernel.project_dir'),
+                '/_profiler/open?file=%%f&amp;line=%%l#line%%l',
```
Comment (xabbuh) [inline src/Symfony/Bundle/WebProfilerBundle/Resources/config/profiler.php:70]: Shouldn't `&amp;` become `&`?

File: src/Symfony/Bundle/WebProfilerBundle/Resources/config/profiler.php [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            ])
+
+        ->set('debug.file_link_formatter.url_format', 'string')
+            ->factory([FileLinkFormatter::class, 'generateUrlFormat'])
+            ->args([
+                service('router'),
+                '_profiler_open_file',
+                '?file=%%f&amp;line=%%l#line%%l',
```
Comment (xabbuh) [inline src/Symfony/Bundle/WebProfilerBundle/Resources/config/profiler.php:78]: Same here?

File: src/Symfony/Bundle/WebProfilerBundle/Resources/config/profiler.php (hunk 1 above)
Comment (Tobion) [reply src/Symfony/Bundle/WebProfilerBundle/Resources/config/profiler.php:70]: i think so too as `&amp;` was just for escaping in xml

Comment (jschaedl) [reply src/Symfony/Bundle/WebProfilerBundle/Resources/config/profiler.php]: Both places are fixed.

**Review by xabbuh — APPROVED** [review]

File: src/Symfony/Bundle/WebProfilerBundle/Resources/config/toolbar.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+namespace Symfony\Component\DependencyInjection\Loader\Configurator;
+
+use Symfony\Bundle\WebProfilerBundle\EventListener\WebDebugToolbarListener;
+
+return static function (ContainerConfigurator $container) {
+    $container->services()
+
+        ->set('web_profiler.debug_toolbar', WebDebugToolbarListener::class)
+            ->tag('kernel.event_subscriber')
```
Comment (Tobion) [inline src/Symfony/Bundle/WebProfilerBundle/Resources/config/toolbar.php:20]: please move `tag` after `arg` to be consistent will all the other migrations

File: src/Symfony/Bundle/WebProfilerBundle/Resources/config/profiler.php [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+                service('profiler')->nullOnInvalid(),
+            ])
+
+        ->set('web_profiler.csp.handler', ContentSecurityPolicyHandler::class)
+            ->args([
+                inline_service(NonceGenerator::class),
+            ])
+
+        ->set('twig.extension.webprofiler', WebProfilerExtension::class)
+            ->tag('twig.extension')
```
Comment (Tobion) [inline src/Symfony/Bundle/WebProfilerBundle/Resources/config/profiler.php:58]: same

**Review by Tobion — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/WebProfilerBundle/Resources/config/toolbar.php (hunk 3 above)
Comment (jschaedl) [reply src/Symfony/Bundle/WebProfilerBundle/Resources/config/toolbar.php:20]: done

File: src/Symfony/Bundle/WebProfilerBundle/Resources/config/profiler.php (hunk 4 above)
Comment (jschaedl) [reply src/Symfony/Bundle/WebProfilerBundle/Resources/config/profiler.php]: done

**Review by Tobion — APPROVED** [review]

**Comment by Tobion** [comment]
Thank you @jschaedl.

**Pull request merged**